src/Tests/Microsoft.NET.Build.Tests/ArtifactsOutputPathTests.cs

```diff
@@ -0,0 +1,216 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Microsoft.NET.Build.Tests
+{
+    using ArtifactsTestExtensions;
+
+    public class ArtifactsOutputPathTests : SdkTest
+    {
+        public ArtifactsOutputPathTests(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        (List<TestProject> testProjects, TestAsset testAsset) GetTestProjects(bool useDirectoryBuildProps, [CallerMemberName] string callingMethod = "")
+        {
+            var testProject1 = new TestProject()
+            {
+                Name = "App1",
+                IsExe = true
+            };
+
+            var testProject2 = new TestProject()
+            {
+                Name = "App2",
+                IsExe = true
+            };
+
+            var testLibraryProject = new TestProject()
+            {
+                Name = "Library",
+            };
+
+            testProject1.ReferencedProjects.Add(testLibraryProject);
+            testProject2.ReferencedProjects.Add(testLibraryProject);
+
+            List<TestProject> testProjects = new() { testProject1, testProject2, testLibraryProject };
+
+            foreach (var testProject in testProjects)
+            {
+                testProject.UseArtifactsOutput = true;
+                testProject.UseDirectoryBuildPropsForArtifactsOutput = useDirectoryBuildProps;
+            }
+
+            var testAsset = _testAssetsManager.CreateTestProjects(testProjects, callingMethod: callingMethod, identifier: useDirectoryBuildProps.ToString());
+
+            if (useDirectoryBuildProps)
+            {
+                File.WriteAllText(Path.Combine(testAsset.Path, "Directory.Build.props"),
+                   """
+                    <Project>
+                        <PropertyGroup>
+                            <UseArtifactsOutput>true</UseArtifactsOutput>
+                        </PropertyGroup>
+                    </Project>
+                    """);
+            }
+
+            return (testProjects, testAsset);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ItUsesArtifactsOutputPathForBuild(bool useDirectoryBuildProps)
```

Comment:
Do we have a test for a custom name of  `<ArtifactsPath>foo</ArtifactsPath>`?

Reply:
Not yet.

Reply:
OK, it should be added now.